### DISCUSSION TITLE
Chemprop seems to use very little GPU -- not sure why

### DISCUSSION THREAD

**Discussion by sbembenek18** [chemprop/chemprop#1019] [Q&A] [answered]
We're still on 1.7.1 -- in the process of migrating to v2. 

We're running on: 

Intel 14900KF
Nvidia 4090 
System RAM= 128MB

During both predict and training phases, GPU usage is very light (less than 10% for predict, less than 30% for training and only intermittently, not throughout) -- which doesn't seem right. The number of molecules in each case is 1M. 

I can confirm chemprop sees the GPU, for example chemprop_predict and chemprop_train show --> [--no_cuda] [--gpu {0}]

How much GPU usage does one usually get here? 

We're running Cuda 12.4 and Cudnn 9.4

Chemprop was install into its own conda env per these steps --> 

1) Downloaded 1.7.1 --> https://github.com/chemprop/chemprop/releases/tag/v1.7.1
2) cd chemprop-1.7.1
3) conda env create -f environment.yml
4) conda activate chemprop
5) pip install --root-user-action=ignore -e . (pip version = 24.2)

Additionally, we installed --> 

conda install pytorch torchvision torchaudio pytorch-cuda=12.4 -c pytorch -c nvidia --name chemprop

Thanks for the help!

**Comment by JacksonBurns** [comment, marked as answer]
Hi @sbembenek18 

I'm not sure what you mean when you say '10% usage' - is this memory usage, or volatile utilization?

If it's the former you can increase your batch size to use more GPU memory, though bear in mind that [large batches sizes can negatively impact generalizability](https://www.sciencedirect.com/science/article/pii/S2405959519303455). v2 has a smaller memory footprint because the individual molecule datapoints are smaller, so this will still show up as you upgrade.

If it's the latter, it's not totally unexpected (especially with v1). ChemProp's update operation doesn't work well on GPUs (the number of add operations for each batch of graphs is inconsistent between batches, since the number of edges varies), but v2 uses `scatter` which makes it less inefficient.
